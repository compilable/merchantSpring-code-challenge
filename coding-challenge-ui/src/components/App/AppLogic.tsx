import React, { useState } from "react"
import { User } from "./User"
import { OverdueOrder } from "../OverdueOrderTable/OverdueOrder"
const API_ENDPOINT = "http://localhost:8080/"


const AppLogic = () => {

  const [user, setUser] = useState<User | null>(null);
  const [overdueOrders, _setOverdueOrders] = useState<Array<OverdueOrder>>([]);

  const setOverdueOrders = (overdueOrders: Array<OverdueOrder>) => {
    overdueOrders.map((overdueOrder) => {
      return (overdueOrder.country = overdueOrder.country.toLowerCase().substring(0, 2))
    });
    _setOverdueOrders(overdueOrders)
  };

  const logError = (error:any)=>{
    console.error(`unable to load the data from the server, please check the connection to : ${API_ENDPOINT}`)
  }

  const makeServiceCall = (endpoint:string)=>{
    fetch(API_ENDPOINT + endpoint)
    .then((results) => results.json())
    .then((data) => {
      if(endpoint==="user"){
        setUser(data)
      }else if(endpoint==="overdueOrders"){
        setOverdueOrders(data)
      }
      
    }, (error) => {
      if (error) {
        logError(error)
      }
    });
  }

  React.useEffect(() => {
    makeServiceCall("user")
    makeServiceCall("overdueOrders")
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);


  return { user, overdueOrders, setOverdueOrders, setUser }

}

export default AppLogic