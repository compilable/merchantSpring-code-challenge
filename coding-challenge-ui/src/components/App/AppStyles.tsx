
import styled from "styled-components"

export const AppWrapper = styled.div`
    height: 100vh;
    width: 100vw;
    background-color: #cccccc;`
  ;

export const AppHeader = styled.header`
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0rem 2rem;
`;

export const HeaderText = styled.h1`
  font-family: "Roboto", sans-serif;
`;

export const Username = styled.span`
  font-family: "Roboto", sans-serif;
`;
