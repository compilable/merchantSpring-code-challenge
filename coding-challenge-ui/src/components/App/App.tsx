import OverdueOrderTable from "../OverdueOrderTable/OverdueOrderTable"
import {AppWrapper, AppHeader, HeaderText, Username} from "./AppStyles"
import AppLogic from "./AppLogic"

const App = () => {

  const{user,overdueOrders}= AppLogic()
  return (
    <AppWrapper>
      <AppHeader>
        <HeaderText>Analytics Dashboard</HeaderText>
        <Username>Welcome, {user ? user.firstName : "Guest"}!</Username>
      </AppHeader>
      {
        <OverdueOrderTable heading="Overdue Orders" orders={overdueOrders} />
      }
    </AppWrapper>
  );
};

export default App;
