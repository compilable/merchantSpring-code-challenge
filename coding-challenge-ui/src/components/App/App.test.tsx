
import { render } from '@testing-library/react'
import App from './App';
import { shallow, mount, ReactWrapper } from 'enzyme';
import toJSON from "enzyme-to-json"
import { act } from "react-dom/test-utils"

import AppLogic from "./AppLogic"

describe('test user data recived', () => {
  /*const{user,setUser}= AppLogic()
  beforeEach(() => {
    setUser({
      firstName: "Jane",
      lastName: "Doe",
      email: "janedoe@email.com",
      id: 1,
    });
  });
  it('should SET the user data', () => {
    console.log(user)
  });*/
//  expect(toJSON(wrapper)).toMatchSnapshot();
});


test("has the main component loaded", () => {
  const wrapper = shallow(<App />);
  expect(toJSON(wrapper)).toMatchSnapshot()
});

test("has OverdueOrderTable component loaded", () => {
  const wrapper = shallow(<App />);
  expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper.text().includes("OverdueOrderTable"))
});

test('renders app title', () => {
  const { getByText } = render(<App />);
  const headerText = getByText("Analytics Dashboard");
  expect(headerText)
});

test('renders user name', () => {
  const { getByText } = render(<App />);
  const headerText = getByText("Welcome, Guest!");
  expect(headerText)
});
