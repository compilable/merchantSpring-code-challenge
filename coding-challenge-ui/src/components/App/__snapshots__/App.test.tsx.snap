// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has OverdueOrderTable component loaded 1`] = `
<styled.div>
  <styled.header>
    <styled.h1>
      Analytics Dashboard
    </styled.h1>
    <styled.span>
      Welcome, 
      Guest
      !
    </styled.span>
  </styled.header>
  <OverdueOrderTable
    heading="Overdue Orders"
    orders={Array []}
  />
</styled.div>
`;

exports[`has the main component loaded 1`] = `
<styled.div>
  <styled.header>
    <styled.h1>
      Analytics Dashboard
    </styled.h1>
    <styled.span>
      Welcome, 
      Guest
      !
    </styled.span>
  </styled.header>
  <OverdueOrderTable
    heading="Overdue Orders"
    orders={Array []}
  />
</styled.div>
`;
